probes/mqlight-probe.js

```diff
@@ -13,166 +13,239 @@
-var url = require('url');
 var am = require('../');
 
 /**
  * Probe to instrument the MQLight npm client
  */
 function MQLightProbe() {
-	Probe.call(this, 'mqlight');
+  Probe.call(this, 'mqlight');
 }
 util.inherits(MQLightProbe, Probe);
 
 MQLightProbe.prototype.attach = function(name, target) {
-	var that = this;
-	if( name != 'mqlight' ) return target;
-	if(target.__probeAttached__) return;
-	target.__probeAttached__ = true;
+  var that = this;
+  if (name != 'mqlight') return target;
+  if (target.__probeAttached__) return;
+  target.__probeAttached__ = true;
 
-	// After 'createClient'
-	aspect.after(target, 'createClient', {}, function(target, methodName, createClientArgs, context, rc) {
-		var thisClient = rc; // the MQLight client that was created
-		// Just monitor 'send' for now as not sure what else will be useful
-		var methods = 'send';
-		// Before one of the above methods is called on the client
-		aspect.around(thisClient, methods, function(target, methodName, args, probeData) {
-			// Start the monitoring for the method
-			that.metricsProbeStart(probeData, methodName, args);
-			that.requestProbeStart(probeData, methodName, args);
-			// Advise the callback for the method.  Will do nothing if no callback is registered
-			aspect.aroundCallback(args, probeData, function(target, callbackArgs, probeData){
-				// method has completed and the callback has been called, so end the monitoring
+  // After 'createClient'
+  aspect.after(target, 'createClient', {}, function(
+    target,
+    methodName,
+    createClientArgs,
+    context,
+    rc
+  ) {
+    var thisClient = rc; // the MQLight client that was created
+    // Just monitor 'send' for now as not sure what else will be useful
+    var methods = 'send';
+    // Before one of the above methods is called on the client
+    aspect.around(
+      thisClient,
+      methods,
+      function(target, methodName, args, probeData) {
+        // Start the monitoring for the method
+        that.metricsProbeStart(probeData, methodName, args);
+        that.requestProbeStart(probeData, methodName, args);
+        // Advise the callback for the method.  Will do nothing if no callback is registered
+        aspect.aroundCallback(args, probeData, function(
+          target,
+          callbackArgs,
+          probeData
+        ) {
+          // method has completed and the callback has been called, so end the monitoring
 
-				//Call the transaction link with a name and the callback for strong trace
-                var callbackPosition = aspect.findCallbackArg(methodArgs);
-                if (typeof(callbackPosition) != 'undefined') {
-                    aspect.strongTraceTransactionLink('mqlight: ', methodName, methodArgs[callbackPosition]);
-                }
+          // Call the transaction link with a name and the callback for strong trace
+          var callbackPosition = aspect.findCallbackArg(methodArgs); // FIXME(sam) methodArgs does not exist!
```

Comment:
I think just change it if you're happy to do that, or I can do it as a separate PR.  We don't really have the cycles right now to be improving the probes tests unfortunately, we just rely on manual testing by the developer at the time of writing which obviously didn't happen in this case...